[video_player_platform_interface] Platform view support

FirentisTFW opened the pull request:
This PR is a subset of #8237 - it only adds platform interface changes.

Partially resolves [86613](https://github.com/flutter/flutter/issues/86613).

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I [linked to at least one issue that this PR fixes] in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style], or this PR is [exempt from CHANGELOG changes].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

<!-- Links -->
[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[Discord]: https://github.com/flutter/flutter/blob/master/docs/contributing/Chat.md
[linked to at least one issue that this PR fixes]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#overview
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#version
[following repository CHANGELOG style]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#changelog-style
[exempt from CHANGELOG changes]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#changelog
[test-exempt]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#tests

FirentisTFW commented:
@stuartmorgan This PR changes the perception a little bit - we now should use player ID in app-facing package, and have no knowledge about texture ID there. How should I go about renaming `textureId` to `playerId`? `textureId` is currently used for parameter names in many platform interface methods. Then, it's also used for helper variables, in tests, and in other files.

What would be best way to rename it in all packages without breaking the static analysis in the meantime? What kind of changes changes should I add to this PR? Rename it in platform interface? Rename it everywhere? We're currently in an ugly state where we allow using platform views, but we still use `textureId` for parameter names everywhere.

stuartmorgan commented:
I think the best way to do this is probably:
1) In this PR, rename it throughout the platform interface
2) In follow-up PRs (for iOS and Android it should just be part of the PRs that add platform view support), rename it in the implementation packages.
3) When updating app-facing package calls, like `create` and `buildView`, to the new versions, update the name there.

In order to pass CI you'll need to add `// ignore` directives to all the implementation packages as part of this PR, and then remove them as the implementations are updated in step 2.

FirentisTFW commented:
@stuartmorgan As suggested, in this PR I renamed parameters in the platform interface and added ignores to the package implementations. The checks on the CI now fail, but I assume that this could be resolved with `override: no versioning needed` label?

stuartmorgan commented:
The version failures we can override, but the other failure is (by design) not overridable. In this case it's a tool bug, so https://github.com/flutter/packages/pull/8468 will need to land and then this rebased on that; sorry about that.

FirentisTFW commented:
@stuartmorgan I see that the other PR was already merged, so I rebased this one. One check failed, but looks like it's infra failure, so I t think it can be re-run

FirentisTFW commented on packages/video_player/video_player_platform_interface/lib/video_player_platform_interface.dart:
```diff
@@ -44,67 +44,81 @@ abstract class VideoPlayerPlatform extends PlatformInterface {
 
   /// Clears one video.
-  Future<void> dispose(int textureId) {
+  Future<void> dispose(int playerId) {
```
Yes, `playerId` is used for both. The goal is to stop using the texture ID in the app-facing package, and rely only on the player ID (though initially the player ID would be the same as the texture ID for texture-based players, it would just not be explicitly revealed).

hellohuanlin replied:
is the goal to eventually replace texture based implementation? or are we keeping both?

hellohuanlin replied:
> We will keep both for the foreseeable future;

Maybe `playerOrTextureId`?

stuartmorgan replied:
See discussion in the combined PR; texture ID will become an internal implementation detail. Nothing above the platform interface layer actually needs to know the texture ID, all we need is a player ID. Implementations that use textures will track the texture ID internally, just as they track platform view IDs internally if necessary.

stuartmorgan reviewed (approved):
LGTM!
